### DISCUSSION TITLE
TypeError: __init__() takes 1 positional argument but 3 were given

### DISCUSSION THREAD

**Discussion by chriswangky** [Lightning-AI/pytorch-lightning#12831] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
When I was training my model,  I got an error .
Traceback is as follows:
** 
File "/home/datasets/traffic_classification/ml/MCT.py", line 355, in __init__
 self.fusion = Fusion_Block(n, d_model, self.n_layers_s1, self.n_layers_s2, self.n_heads, self.d_ff,
File "/home/datasets/traffic_classification/ml/MCT.py", line 313, in __init__
self.c2trans = CNN_to_Trans(n, d_model)
File "/usr/local/miniconda3/envs/deep_packet/lib/python3.8/site-packages/pytorch_lightning/core/lightning.py", line 92, in __init__
super().__init__(*args, **kwargs)
TypeError: __init__() takes 1 positional argument but 3 were given
**
Here is my code.
```
class CNN_to_Trans(LightningModule):
    def __int__(self, n, d_model):
        super().__init__() 
        self.pooling = nn.AdaptiveAvgPool1d(n)
        self.conv = nn.Conv1d(in_channels=n*d_model, out_channels=d_model, kernel_size=1,stride=1)
        self.norm = nn.LayerNorm(d_model)
        self.act_f = nn.ReLU()
    def forward(self, x):
        x = self.conv(x.transpose(1,2))
        x = self.pooling(x).transpose(1,2)
        x = self.norm(x)
        x = self.act_f(x)
        return x
```

```
class Fusion_Block(LightningModule):
    def __init__(self, n, d_model, n_layers_s1, n_layers_s2, n_heads, d_ff, kernel_size, stage1_med_dim, stage1_out_dim, stage2_out_dim):
        super(Fusion_Block, self).__init__()
        ... ...
        self.c2trans = CNN_to_Trans(n, d_model)
        ... ...

    def forward(...):
        ... ...
```

Thank you!

**Comment by i-aki-y** [comment, marked as answer]
@chriswangky  Hi, is it a typo? `__int__` --> `__init__`

**Reply by chriswangky** [reply]
Thank you! I am a fool.
